File: lw2/task4.pas
PROGRAM WorkWithQueryString(INPUT, OUTPUT);
USES
  DOS;

FUNCTION GetQueryStringParameter(Key: STRING): STRING;
VAR                                                    
 Param, Querystring: STRING;
 PosKey: INTEGER; 
BEGIN {GetQueryStringParameter}
  Querystring := GetEnv('QUERY_STRING');
  PosKey := POS(Key, GetEnv('QUERY_STRING'));
  IF (PosKey <> 0) AND (COPY(Querystring, PosKey + LENGTH(Key), 1) = '=') AND ((COPY(Querystring, PosKey + LENGTH(Key) + 1, 1) <> '&') AND (COPY(Querystring, PosKey + LENGTH(Key) + 1, 1) <> ''))
  //Переделать 
  THEN
    BEGIN
      Param := COPY(Querystring, PosKey + LENGTH(Key) + 1, 255);
      IF POS('&', Param) <> 0
      THEN
        GetQueryStringParameter := COPY(Param, 1, POS('&', Param) - 1)
      ELSE
        GetQueryStringParameter := Param;
    END
 ELSE
    GetQueryStringParameter := 'Not found'
END; {GetQueryStringParameter}

BEGIN {WorkWithQueryString}
  WRITELN('Content-Type: text/plain');
  WRITELN;
  WRITELN('First Name: ', GetQueryStringParameter('first_name'));
  WRITELN('Last Name: ', GetQueryStringParameter('last_name'));
  WRITELN('Age: ', GetQueryStringParameter('age'))
END. {WorkWithQueryString}

File: lw3/task4/task4.php
<?php

header('Content-Type: text/plain');
$firstName = isset($_GET['first_name']) ? $_GET['first_name'] : '...'; //'пробел' (для task5)
$lastName = isset($_GET['last_name']) ? $_GET['last_name'] : '...';
$age = isset($_GET['age']) ? $_GET['age'] : '...';
$email = isset($_GET['email']) ? $_GET['email'] : null;
if (($email !== null) and ($email !== ''))
{
    if (filter_var($email, FILTER_VALIDATE_EMAIL))
    {
        $fEmail = fopen("data/$email.txt", "w");
        fwrite($fEmail, "Email: $email\r\n");
        fwrite($fEmail, "First name: $firstName\r\n");
        fwrite($fEmail, "Last name: $lastName\r\n");
        fwrite($fEmail, "Age: $age");
        fclose($fEmail);  
    }
    else
    {
        echo 'Invalid email entered.';
    }
}
else
{
    echo "Parameter 'email' not found.";
}